3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/AnonIpEnrichmentSpec.scala

```diff
@@ -23,20 +23,35 @@ import org.specs2.matcher.DataTables
-      AnonIpEnrichment(octets).anonymizeIp(ip) must_== expected
+    "SPEC NAME"              || "IP ADDRESS"                    | "ANONYMIZE IPv4 OCTETS" | "ANONYMIZE IPv6 PIECES" | "EXPECTED OUTPUT" |
+      "valid, anonymize 1"   !! "0.23.0.20"                     ! AnonIPv4Octets(1)       ! AnonIPv6Segments(1)     ! "0.23.0.x" |
+      "valid, anonymize 2"   !! "168.192.102.4"                 ! AnonIPv4Octets(2)       ! AnonIPv6Segments(2)     ! "168.192.x.x" |
+      "valid, anonymize 3"   !! "54.242.102.43"                 ! AnonIPv4Octets(3)       ! AnonIPv6Segments(3)     ! "54.x.x.x" |
+      "valid, anonymize 4"   !! "94.15.213.171"                 ! AnonIPv4Octets(4)       ! AnonIPv6Segments(4)     ! "x.x.x.x" |
+      "invalid, anonymize 1" !! "777.2"                         ! AnonIPv4Octets(1)       ! AnonIPv6Segments(1)     ! "777.2" |
+      "invalid, anonymize 4" !! "777.2.23"                      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(4)     ! "x.x.x" |
+      "invalid, anonymize 3" !! "999.123.777.2"                 ! AnonIPv4Octets(3)       ! AnonIPv6Segments(3)     ! "999.x.x.x" |
+      "invalid, anonymize 3" !! "hello;goodbye"                 ! AnonIPv4Octets(3)       ! AnonIPv6Segments(3)     ! "hello;goodbye" |
+      "null, anonymize 2"    !! null                            ! AnonIPv4Octets(2)       ! AnonIPv6Segments(2)     ! null |
+      "empty, anonymize 4"   !! ""                              ! AnonIPv4Octets(4)       ! AnonIPv6Segments(4)     ! "x" |
+      "ipv6, anonymize 1"    !! "4b0c:0:0:0:880c:99a8:4b0:4411" ! AnonIPv4Octets(1)       ! AnonIPv6Segments(1)     ! "4b0c:0:0:0:880c:99a8:4b0:x" |
+      "ipv6, anonymize 2"    !! "4b0c::880c:99a8:4b0:4411"      ! AnonIPv4Octets(2)       ! AnonIPv6Segments(2)     ! "4b0c:0:0:0:880c:99a8:x:x" |
+      "ipv6, anonymize 3"    !! "2605:2700:0:3:0:0:4713:93e3"   ! AnonIPv4Octets(3)       ! AnonIPv6Segments(3)     ! "2605:2700:0:3:0:x:x:x" |
+      "ipv6, anonymize 4"    !! "2605:2700:0:3::4713:93e3"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(4)     ! "2605:2700:0:3:x:x:x:x" |
+      "ipv6, anonymize 5"    !! "2605:2700:0:3::4713:93e3"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(5)     ! "2605:2700:0:x:x:x:x:x" |
+      "ipv6, anonymize 6"    !! "2605:2700:0:3::4713:93e3"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(6)     ! "2605:2700:x:x:x:x:x:x" |
+      "ipv6, anonymize 7"    !! "2605:2700:0:3::4713:93e3"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(7)     ! "2605:x:x:x:x:x:x:x" |
+      "ipv6, anonymize 8"    !! "2605:2700:0:3::4713:93e3"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(8)     ! "x:x:x:x:x:x:x:x" |
+      "ipv6, anonymize 8"    !! "2605:2700::4713:93e3"          ! AnonIPv4Octets(4)       ! AnonIPv6Segments(8)     ! "x:x:x:x:x:x:x:x" |
+      "ipv6, anonymize 8"    !! "2700::4713"                    ! AnonIPv4Octets(4)       ! AnonIPv6Segments(8)     ! "x:x:x:x:x:x:x:x" |
```

Comment:
I've made an arbitrary decision to return IPv6 in not shortened version. It simplifies development and makes it visible that some anonymization took place.

Reply:
👍🏻